backend/routes.js

```diff
@@ -63,6 +63,65 @@ const cidLookupTable = {
   [multicodes.SHELTER_FILE_CHUNK]: 'application/vnd.shelter.filechunk+octet-stream'
 }
 
+const IPV4_ADDR_REGEX = /^(?:[3-9]\d{0,1}|1\d{0,2}|2\d{0,1}|2[0-4]\d|25[0-5])(?:\.(?:0|[3-9]\d{0,1}|1\d{0,2}|2\d{0,1}|2[0-4]\d|25[0-5])){3}$/
+const IPV6_SEGMENT_REGEX = /^[0-9a-fA-F]{1,4}$/
+const limiterKey = (ip: string) => {
+  if (IPV4_ADDR_REGEX.test(ip)) {
```

Comment:
Is this the fastest way to do this or would something like `ip.split('.').each(x => 0 < parseInt(x) <= 255` be faster?

Reply:
Not sure. It might or might not be faster but a regex seems like the right tool for this. Also, your example will accept invalid addresses such as `1foo.2bar.3baz.4`, addresses of dubious validity such as `000000001.2.3.4`, and then also the length needs to be checked (there should be 4 segments). It'll also accept `0.x.x.x` which are _not_ valid addresses (although this is a minor issue)